base/inference.jl

```diff
@@ -1400,10 +1398,6 @@ function abstract_call_gf_by_type(f::ANY, atype::ANY, sv::InferenceState)
         add_mt_backedge(ftname.mt, argtype, sv)
         update_valid_age!(min_valid[1], max_valid[1], sv)
     end
-    if isempty(applicable)
-        # TODO: this is needed because type intersection is wrong in some cases
-        return Any
```

Comment:
When I proposed this in #20866 you seemed rather skeptical. Now it's my turn to be skeptical :smile:
There's no doubt we want this in the long run, but I now think it's still too early because type intersection is still too buggy. (We don't need the extra crashes produced by this to raise extra awareness of these bugs, they are still easy enough to find anyway.)

And I wouldn't want this PR to be delayed due to open type intersection bugs.

For the record: This is responsible for the crash in the arrayops test.